### DISCUSSION TITLE
組長要如何acp到football-bpj1上

### DISCUSSION THREAD

**Discussion by Maskie-Razzio** [mdecd2023/football-apj1#5] [Q&A]

**Comment by scrum-2** [comment]
原始問題:

是要重新git clone一個新網頁嗎?
還是要fork下來，如果要fork的話，之後要怎們acp?

不是要你刪除原先的提問, 只是強調, 針對每一項術語, 要明確知道其意涵, 以及使用時機, 這樣提問起來才會切中要點, 也會比較容易找到答案.

git clone 的時機通常有兩個, 一個是想看看倉儲內容在近端執行的情況, 可能對該倉儲沒有管理權, 而另一個目的則是希望改版.

若 git clone 後要改版, 則可以透過 https 或 ssh 協定, 以 https 則需要設定 token, 若以 ssh 則需要 key pairs 設定.

另外 fork 是將別人的倉儲複製一份, 放在自己的帳號下, 且與原始倉儲維持一定的關係 (這個關係若原始倉儲刪除或改為 private, 則會消失, 但 fork 後的倉儲內容仍會存在), 因為 fork 後在自己的帳號下, 所以就可以用自己的權限對其改版. 之後, 若要將此 forked repo 改版內容回饋到原始倉儲, 只能建立 pull requests, 請求原始倉儲的管理者審查是否要將改版內容合併回去.

而 acp 在 windows 其實是 acp.bat 是連續執行 git add ., git commit -m 與 git push 等三個指令, 假如 git commit -m 沒有設定 .git config 中的 user.name 與 user.email, 就無法執行, 而 git push 則需視採 https 或 ssh 權限, 各有不同的設定流程.

**Reply by Maskie-Razzio** [reply]
好，謝謝老師
